src/Facets/DeBridgeDlnFacet.sol

```diff
@@ -0,0 +1,182 @@
+    struct DeBridgeDlnData {
+        address receivingAssetId;
+        bytes receiver;
+        uint256 minAmountOut;
+    }
+
+    /// Events ///
+
+    event DlnOrderCreated(bytes32 indexed orderId);
+
+    event BridgeToNonEVMChain(
+        bytes32 indexed transactionId,
+        uint256 indexed destinationChainId,
+        bytes receiver
+    );
+
+    /// Constructor ///
+
+    /// @notice Constructor for the contract.
+    /// @param _dlnSource The address of the DLN order creation contract
+    constructor(IDlnSource _dlnSource) {
+        dlnSource = _dlnSource;
+    }
+
+    /// External Methods ///
+
+    /// @notice Bridges tokens via DeBridgeDLN
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _deBridgeDlnData Data specific to DeBridgeDLN
+    function startBridgeTokensViaDeBridgeDln(
+        ILiFi.BridgeData memory _bridgeData,
+        DeBridgeDlnData calldata _deBridgeDlnData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+    {
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+        _startBridge(_bridgeData, _deBridgeDlnData);
+    }
+
+    /// @notice Performs a swap before bridging via DeBridgeDLN
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _swapData An array of swap related data for performing swaps before bridging
+    /// @param _deBridgeDlnData Data specific to DeBridgeDLN
+    function swapAndStartBridgeTokensViaDeBridgeDln(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        DeBridgeDlnData calldata _deBridgeDlnData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+    {
+        uint256 fee = dlnSource.globalFixedNativeFee();
+        address assetId = _bridgeData.sendingAssetId;
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender),
+            LibAsset.isNativeAsset(assetId) ? 0 : fee
+        );
+        _startBridge(_bridgeData, _deBridgeDlnData);
+    }
+
+    /// Internal Methods ///
+
+    /// @dev Contains the business logic for the bridge via DeBridgeDLN
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _deBridgeDlnData Data specific to DeBridgeDLN
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        DeBridgeDlnData calldata _deBridgeDlnData
+    ) internal {
+        uint256 fee = dlnSource.globalFixedNativeFee();
```

Comment:
This is done twice for swapAnd.. cases. Or pass it into _startBridge() as parameter (and pull it in both startBridgeTokensViaDeBridgeDln and swapAndStartBridgeTokensViaDeBridgeDln) to avoid having to do it twice.
 
Or could even consider to move this into backend and make it a parameter of the DeBridgeDlnData